junit-platform-launcher/src/main/java9/org/junit/platform/launcher/listeners/FlightRecordingListener.java

```diff
@@ -112,27 +120,38 @@ public void executionFinished(TestIdentifier test, TestExecutionResult result) {
 	@Name("org.junit.Test")
 	@Label("Test")
 	static abstract class TestEvent extends Event {
-
 		@Label("Unique Id")
 		String uniqueId;
-
 		@Label("Display Name")
 		String displayName;
 	}
 
 	@Name("org.junit.DynamicTestRegistration")
+	@Label("Dynamic Test Registration")
 	static class DynamicTestEvent extends TestEvent {
 	}
 
 	@Name("org.junit.SkippedTest")
+	@Label("Skipped Test")
 	static class SkippedTestEvent extends TestEvent {
 		@Label("Reason")
 		String reason;
 	}
 
 	@Name("org.junit.TestExecution")
+	@Label("Test")
 	static class TestExecutionEvent extends TestEvent {
 		@Label("Result")
 		String result;
 	}
+
+	@Category("JUnit")
+	@Name("org.junit.ReportEntry")
+	@Label("Report Entry")
+	static class ReportEntryEvent extends Event {
```

Comment:
How often is a report emitted, once per test? What would it typically contain? There may be other ways to structure the data, but I need more information.

Reply:
This is an event triggered by a user or an extension within the lifecycle of a test.

Here's an example:
https://github.com/junit-team/junit5/blob/19785134f95ba100346eb2d8b0c24f707535ebe4/documentation/src/test/java/example/TestReporterDemo.java#L25-L42

Reply:
Here's the API documentation for `TestReporter`: https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/TestReporter.html